### ISSUE TITLE
NuGet/Agent version mismatch results in TypeLoadException

### ISSUE THREAD

**Issue by gshackles** [DataDog/dd-trace-dotnet#550]
**Describe the bug**
When our app (ASP.NET Core 2.2) was referencing the 1.8 version of the NuGet package, and the agent on the host was running version 1.7, requests to the app would result in `TypeLoadExceptions` like this when Datadog was attempting to instrument things:

```System.TypeLoadException: Could not load type 'Datadog.Trace.ClrProfiler.Integrations.AdoNetIntegration' from assembly 'Datadog.Trace.ClrProfiler.Managed, Version=1.8.0.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb'.
     at System.Data.Common.DbCommand.ExecuteReader()
     at Npgsql.NpgsqlCommand.ExecuteReader()
     at Npgsql.PostgresDatabaseInfo.LoadBackendTypes(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
     at Npgsql.PostgresDatabaseInfo.LoadPostgresInfo(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
     at Npgsql.PostgresDatabaseInfoFactory.Load(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
     at Npgsql.NpgsqlDatabaseInfo.Load(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async)
     at Npgsql.NpgsqlConnector.LoadDatabaseInfo(NpgsqlTimeout timeout, Boolean async)
     at Npgsql.NpgsqlConnector.Open(NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
     at Npgsql.ConnectorPool.AllocateLong(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken)
     at System.Threading.Tasks.ValueTask`1.get_Result()
     at Npgsql.NpgsqlConnection.<>c__DisplayClass32_0.<<Open>g__OpenLong|0>d.MoveNext()
  --- End of stack trace from previous location where exception was thrown ---
     at Npgsql.NpgsqlConnection.Open()
     at Dapper.SqlMapper.ExecuteCommand(IDbConnection cnn, CommandDefinition& command, Action`2 paramReader) in C:\projects\dapper\Dapper\SqlMapper.cs:line 2827
     at Dapper.SqlMapper.ExecuteImpl(IDbConnection cnn, CommandDefinition& command) in C:\projects\dapper\Dapper\SqlMapper.cs:line 570
     at Dapper.SqlMapper.Execute(IDbConnection cnn, String sql, Object param, IDbTransaction transaction, Nullable`1 commandTimeout, Nullable`1 commandType) in C:\projects\dapper\Dapper\SqlMapper.cs:line 443
     at Olo.FraudService.DbConnectionWrapper.Execute(String sql, Object param, IDbTransaction transaction, Nullable`1 commandTimeout, Nullable`1 commandType) in D:\BuildAgent\work\6ed81a1db467b88b\src\IDbConnectionWrapper.cs:line 41
```

Is this something that could be handled more gracefully, in order to avoid having to coordinate things pretty carefully when we go to upgrade the trace agent version?

**To Reproduce**
1. Install trace agent with version 1.8
2. Run ASP.NET Core app with 1.7 version of NuGet
3. Make request to app that makes an ADO.NET call

**Expected behavior**
Either the instrumentation still works normally, or it degrades gracefully with a warning somewhere.

**Runtime environment (please complete the following information):**
- Instrumentation mode: [e.g. automatic with msi installer or manual with NuGet package]
- Tracer version: 1.8
- OS: Windows
- CLR: .NET Core 2.2

**Comment by andrewsouthard1** [comment]
Hey @gshackles thanks again for reporting this issue. As an update we are working on implementing a safety mechanism to ensure this doesn't happen here: https://github.com/DataDog/dd-trace-dotnet/pull/567

**Comment by zacharycmontoya** [comment]
This issue has been fixed in the profiler in #570 and will be included in the next release, which we're planning to roll out shortly.

**Comment by lucaspimentel** [comment]
The fix for this issue is included in [release 1.10.0](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.10.0).

Note that the `Datadog.Trace.ClrProfiler.Managed` NuGet package is no longer required for instrumentation in .NET Core and will be removed soon. To access core classes (e.g. `Tracer`, `TracerSettings`, `Span`) for configuration or manual instrumentation, only package `Datadog.Trace` is needed.
